### PR TITLE
feat: support KUBECTL env var to override kubectl CLI

### LINKED ISSUE #216: Support overriding the kubectl command used by validations
ISVs with their own Kubernetes distributions (e.g., OpenShift) may use CLI tools other than `kubectl`. The validation suite should provide a way for ISVs to specify a custom kubectl-compatible command that is respected by both the shell stubs and the Python validation tests.

Today, kubectl detection is hardcoded to a small set of known providers in `isvtest/core/k8s.py` (`get_kubectl_command()`) and the shell stubs (`setup.sh`, `_common.sh`).

### PR DESCRIPTION
**Pull request by mresvanis**
## Summary

- Add `KUBECTL` environment variable support to `get_kubectl_command()` in Python: parses with `shlex.split()`, validates binary on PATH via `shutil.which()`, falls through to existing `K8S_PROVIDER` detection for empty/whitespace values
- Add inline `${KUBECTL:-}` guard to all four k8s shell stubs (`setup.sh`, `setup_microk8s.sh`, `setup_k3s.sh`, `setup_minikube.sh`) so an env-provided value skips auto-detection
- Document `KUBECTL` in `AGENTS.md` env table and `k8s.yaml` config header

Closes #216

## Test plan

- [x] 11 unit tests covering all edge cases: simple override, whitespace stripping, multi-token prefix, quoted paths with spaces, precedence over `K8S_PROVIDER`, empty/whitespace-only/empty-quoted fallthrough, `get_kubectl_base()` round-trip, binary-not-on-PATH error
- [x] Manual smoke test — verify the override is picked up end-to-end without a cluster:
  ```
  KUBECTL=oc uv run --project isvtest python -c \
      "from isvtest.core.k8s import run_kubectl; r = run_kubectl(['version', '--client']); print(r.stdout)"
  ```
- [x] Manual: verify unset `KUBECTL` preserves existing detection behavior on vanilla k8s

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * KUBECTL environment override: specify a kubectl-compatible CLI (including multi-token commands and quoted paths) that takes precedence over auto-detection; clearer guidance shown when none is found.

* **Documentation**
  * Updated autodetection guidance and docs to explain KUBECTL usage and precedence.

* **Tests**
  * Added tests covering override precedence, parsing, fallbacks, and error handling.

### CODE REVIEW

File: isvctl/configs/stubs/k8s/setup.sh [hunk 1]
```diff
@@ -19,12 +19,14 @@
...
+if [[ "${KUBECTL:-}" =~ [^[:space:]] ]]; then
+    :  # already set from environment; skip detection
+elif command -v kubectl &> /dev/null; then
     KUBECTL="kubectl"
 elif command -v microk8s &> /dev/null; then
     KUBECTL="microk8s kubectl"
 else
-    echo "Error: Neither kubectl nor microk8s found" >&2
+    echo "Error: Neither kubectl nor microk8s found. Set KUBECTL to override." >&2
     exit 1
```
[the remark this answers is not in the document]
Comment (mresvanis) [reply isvctl/configs/stubs/k8s/setup.sh:30]: Every error path across all stubs consistently writes plain text to stderr and exits non-zero. Introducing a JSON error object on stdout in just this one spot would be inconsistent with the established convention and could confuse callers that distinguish success (JSON on stdout) from failure (message on stderr + non-zero exit).

File: isvctl/configs/stubs/k8s/setup_minikube.sh [hunk 2]
```diff
@@ -17,13 +17,16 @@
 
 set -eo pipefail
 
-if ! command -v kubectl &> /dev/null; then
-    echo "Error: kubectl not found (minikube should configure it automatically)" >&2
+# Detect kubectl command
+if [[ -n "${KUBECTL:-}" ]]; then
+    :  # already set from environment; skip detection
+elif command -v kubectl &> /dev/null; then
```
[the remark this answers is not in the document]
Comment (mresvanis) [reply isvctl/configs/stubs/k8s/setup_minikube.sh]: This should have been addressed, please re-check.

**Comment by mresvanis** [comment]
@coderabbitai resume

**Review by abegnoche — APPROVED** [review]
